tests/unit/test_cli.py

```diff
@@ -211,3 +214,63 @@ def testReferenceBasesListArguments(self):
         self.assertEqual(args.end, 2)
         self.assertEquals(args.outputFormat, "fasta")
         self.assertEquals(args.runner, cli.ListReferenceBasesRunner)
+
+
+class TestOutputFormats(unittest.TestCase):
+    """
+    Tests the different output formats of the cli
+    """
+    class FakeArgs(object):
+        def __init__(self, outputFormat='text'):
+            self.outputFormat = outputFormat
+            self.id = 'id'
+            self.key = 'key'
+            self.baseUrl = 'baseUrl'
+            self.verbose = 'verbose'
+
+    class FakeObject(protocol.ProtocolElement):
+
+        __slots__ = ['id', 'name']
+
+        def __init__(self):
+            self.id = 'id'
+            self.name = 'name'
+
+    def _getRunPrintMethodCalls(self, runner):
+        printCalls = []
+        with mock.patch('__builtin__.print') as printMethod:
+            printMethod.side_effect = \
+                lambda *args, **kwargs: printCalls.append((args, kwargs))
+            runner.run()
+        return printCalls
+
+    def testListReferenceBasesFasta(self):
+        args = self.FakeArgs('fasta')
```

Comment:
Mocks are neat, but I find them hard to follow and (particularly) maintain sometimes. Here we're violating some interface rules by accessing the ``_client`` attribute, and we also need to know the detailed interface of  the ``listReferenceBases`` method. Depending on the internal structures in the class makes it difficult to change things, as we don't know what tests we're going to break, or necessarily how to fix the tests that have been broken afterwards. I hit a lot of this in the last refactoring pass.

I would _much_ rather depend on the external, advertised interface for tests. Here for example, why not use a local simulated backend, and capture the output? We can then compare the fasta output with the truth that we get from the backend.